src/urllib3/poolmanager.py

```diff
@@ -76,6 +71,18 @@ class InvalidProxyConfigurationWarning(HTTPWarning):
 #: The namedtuple class used to construct keys for the connection pool.
 #: All custom key schemes should include the fields in this key at a minimum.
 PoolKey = collections.namedtuple("PoolKey", _key_fields)
+_proxy_config_fields = ("ssl_context", "allow_insecure_proxy", "destination_scheme")
+
+
+class ProxyConfig:
```

Comment:
I'm not loving having an additional proxy config object here. The `allow_insecure_proxy` option shouldn't (?) be needed, if a user makes a request with `scheme=https, proxy_scheme=http` then they're opting in to an insecure proxy. Might not be "obvious" to some users but we can document this?

I don't think we lose a lot by having a singular `SSLContext` object shared between Proxy TLS and Connection TLS as any non-local HTTPS proxies are either corporate proxies that are providing their own bundle and MITM connections anyways or are services that have domain names and valid certificates that'd work with certifi, etc.
Am I thinking about this correctly?

Reply:
We actually do need a second SSLContext object for our internal use case :(. I'll explain more in the bottom.

Reply:
Per below it'd be great to make this object a tuple for immutability.